std-bits/database/src/main/java/org/enso/database/nativeimage/SqliteJdbcPatchedFeature.java

```diff
@@ -0,0 +1,103 @@
+package org.enso.database.nativeimage;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeClassInitialization;
+import org.graalvm.nativeimage.hosted.RuntimeJNIAccess;
+import org.sqlite.BusyHandler;
+import org.sqlite.Collation;
+import org.sqlite.Function;
+import org.sqlite.ProgressHandler;
+import org.sqlite.SQLiteJDBCLoader;
+import org.sqlite.core.DB;
+import org.sqlite.core.NativeDB;
+import org.sqlite.jdbc3.JDBC3DatabaseMetaData;
+import org.sqlite.util.LibraryLoaderUtil;
+import org.sqlite.util.OSInfo;
+import org.sqlite.util.ProcessRunner;
+
+/**
+ * Replacement of {@code org.sqlite.nativeimage.SqliteJdbcFeature}. Is almost a direct copy of that
+ * feature, but does not include native libraries inside resources.
+ */
+public class SqliteJdbcPatchedFeature implements Feature {
+
+  @Override
+  public void beforeAnalysis(BeforeAnalysisAccess access) {
```

Comment:
This is unfortunate. Can we send a patch upstream to avoid this copy next time (when we upgrade to new version of SQLite)?

Reply:
It will not work for upstream. This patched feature expects our native library location, which is specific for us. The upstream `SqliteJdbcFeature` extracts some native library from JAR resources and registers it as a resource to Native Image. I don't understand what exactly are you suggesting to change in upstream?